File: libPrjPython/Film.py
from Media import Media

class Film(Media):
    
    def __init__(self, callno, title, subject, director, notes, year):
        super(Film, self).__init__(callno, title, subject, notes)
        self.director = director
        self.year = year
    
    def display(self):
       print "---------------Film Details-------------------------------------------------------------------------"
       super(Film, self).display()
       print "Director             : ", self.director
       print "Notes                : ", self.notes
       print "Year                 : ", self.year
       print "\n"

    def compare_other(self,srch_other):
        return srch_other in self.director or srch_other in self.notes or srch_other in self.year

File: librarySystem/Film.cpp
#include <iostream>
#include "Film.h"
#include "Media.h"
#include <sstream>

using namespace std;

Film::Film(string &cno, string &titl, string &sub, string &dir, string &note, string &yr)
  : Media(cno, titl, sub, note), director(dir), year(yr)
{
}

Film::~Film()
{
}

void Film::display() const
{
	cout<<"--------------Film Details----------------------------------------"<<endl;
	Media::display();
	cout<<"Director---------------"<<this->director<<endl;
	cout<<"Notes------------------"<<this->notes<<endl;
	cout<<"Year-------------------"<<this->year<<endl;
	cout<<"------------------------------------------------------------------"<<endl;
}

bool Film::compare_other(const string &other) const
{
	if((this->director.find(other) != std::string::npos) ||
		(this->notes.find(other) != std::string::npos) ||
		(this->year.find(other) != std::string::npos))
	{
		return true;
	}
	else
		return false;
}
